### PR TITLE
Improve performance of unpack_differentiae

### PR DESCRIPTION
**Pull request by rodsan0**
Refactor function to use `numpy.frombuffer` when bit width is one of 8, 16, 32, or 64.

### CODE REVIEW

File: hstrat/serialization/_unpack_differentiae.py [hunk 1]
```diff
@@ -10,23 +11,25 @@ def unpack_differentiae(
     packed_differentiae: str,
     differentia_bit_width: int,
 ) -> typing.Iterable[int]:
-    """Unpack a sequence of differentiae from into a compact, concatenated
-    base 64 representation.
+    """Unpack a compact, concatenated base 64 representation into
+    a sequence of differentiae.
```
Comment (mmore500) [inline hstrat/serialization/_unpack_differentiae.py:15]: into a sequence with each represented as a distinct integer.

File: hstrat/serialization/_unpack_differentiae.py [hunk 2]
```diff
@@ -10,23 +11,25 @@ def unpack_differentiae(
...
     """
     _bytes = b64decode(packed_differentiae)
-    bits = BitArray(bytes=_bytes)
 
-    # if padding bits are possible, first byte tells how many are required
-    if differentia_bit_width % 8:
+    if differentia_bit_width in [8, 16, 32, 64]:
+        # bit width is a multiple of 8 -- no padding required
+        dt = np.dtype(f">u{differentia_bit_width // 8}")
```
Comment (mmore500) [inline hstrat/serialization/_unpack_differentiae.py:21]: name the floor division result num_bytes and give a comment explaining this numpy dtytoe construtor syntax, maybe link to numpy docs; explain > too

File: hstrat/serialization/_unpack_differentiae.py [hunk 3]
```diff
@@ -10,23 +11,25 @@ def unpack_differentiae(
...
-    else:
-        valid_bits = bits[:]
 
-    assert len(valid_bits) % differentia_bit_width == 0
+        assert len(valid_bits) % differentia_bit_width == 0
 
-    for chunk in iter_chunks(valid_bits, differentia_bit_width):
-        yield chunk.uint
+        for chunk in iter_chunks(valid_bits, differentia_bit_width):
+            yield chunk.uint
```
Comment (mmore500) [inline hstrat/serialization/_unpack_differentiae.py:35]: Split out the bitarray and numpy implementations into sep functions  within same file private leading underscore

**Review by mmore500 — APPROVED** [review]
Once changes made, good to merge

File: hstrat/serialization/_unpack_differentiae.py (hunk 2 above)
Comment (rodsan0) [reply hstrat/serialization/_unpack_differentiae.py]: Done in 120a3e37

File: hstrat/serialization/_unpack_differentiae.py (hunk 3 above)
Comment (rodsan0) [reply hstrat/serialization/_unpack_differentiae.py]: Done in fb47b915

File: hstrat/serialization/_unpack_differentiae.py (hunk 1 above)
Comment (rodsan0) [reply hstrat/serialization/_unpack_differentiae.py]: Changed in 5eab3550

**Pull request merged**